keps/sig-cluster-lifecycle/kubeadm/20191229-kubeadm-move-out-of-kk.md

```diff
@@ -0,0 +1,594 @@
+https://github.com/neolit123/kubeadm/blob/66b94f324b63005589bab7132193e8031001ba57/kubeadm/go.mod
+
+The current proposal is to manually update these dependencies right
+after each release or during a release cycle if something is critical.
+The k/k issue tracker and Slack will be the source of signal if something
+has to be updated.
+
+##### Scenarios:
+
+###### A bug in k8s.io/[some-library] is found and fixed in /staging
+or /vendor of k/k
+
+The kubeadm maintainers need to watch for such changes or bug reports
+and update the kubeadm go.mod file. Running the change in CI will confirm
+that it is not regressing.
+
+###### The libraries in k/k/staging have a lot of changes since the
+last k8s release
+
+The kubeadm maintainers need to pick commits to update the kubeadm go.mod
+file to. Running the changes in CI will confirm that they are not regressing.
+
+
+###### Kubernetes released with bug fixes in dependency X, that kubeadm
+did not include
+
+Kubeadm should consider backporting a dependency update depending on the
+severity of the issue.
+If the issue is low severity evaluate if the the backport can be skipped.
+If the issue is high servery consider a backport.
+
+### Risks and Mitigations
+
+#### A portion of the kubeadm backend changes and breaks users
+
+As mentioned in "User story 3", the kubeadm maintainers
+will give no guarantees about the kubeadm backend until it is
+announced as stable. README files in the kubeadm repository
+should denote that.
+
+It is out of scope for this proposal to try to solve the problem.
+
+#### The post-submit job for tagging/branching k/kubeadm fails
+
+In cases the post-submit job fails, maintainers should be able
+to apply the required changes manually. Sufficient number of
+maintainers should have access to perform these actions.
+
+Close to a release, coordination with SIG Release will be required,
+to make sure that the exact, desired SHA of kubeadm is released.
+
+#### A unwanted commit gets tagged in the kubeadm repository
+
+If the post-submit job is soon to tag a commit in k/kubeadm
+for release (e.g. `v1.18.0`), there is a chance that a PR in k/kubeadm
+can merge before this happens and a commit that is not desired
+receives the tag.
+
+This is a mistake that the kubeadm maintainers should not allow.
+Close to a release, there is code freeze at effect, so PRs should
+pass the proper review and possibly be postponed for the next release.
+
+While kubeadm is already quite low volume in commits, coordination with
+SIG Release might be required to know when the release button is pushed
+and to prevent such problems.
+
+#### The kubeadm go.mod includes outdated components
+
+Once kubeadm is decoupled from k/k it will no longer benefit
+from all the updates and security fixes that its components under
+"staging" see. The imports in the kubeadm go.mod will need
+to be updated on a regular basis to mitigate CVEs and critical bugs.
+
+Tools such as https://docs.renovatebot.com/golang/ were
```

Comment:
Or Dependabot (although its Go support is in alpha).

Reply:
i'm not convinced either of these tools will fit our needs.

Reply:
let's investigate after the move.